testsuite/cluster-test/src/health/fullnode_check.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use crate::{
+    cluster::Cluster,
+    instance::Instance,
+    instance::validator_index,
+    health::{HealthCheck, HealthCheckContext},
+};
+use async_trait::async_trait;
+use futures::future::join_all;
+use once_cell::sync::Lazy;
+use std::collections::HashMap;
+use std::env;
+
+pub static THRESHOLD: Lazy<i64> = Lazy::new(||
+    if let Ok(v) = env::var("FULL_NODE_HEALTH_THRESHOLD") {
+        v.parse().expect("Failed to parse FULL_NODE_HEALTH_THRESHOLD")
+    } else {
+        20000_i64
+    });
+
+pub struct FullNodeHealthCheck {
+    cluster: Cluster,
+}
+
+impl FullNodeHealthCheck {
+    pub fn new(cluster: Cluster) -> FullNodeHealthCheck {
+        Self { cluster }
+    }
+}
+
+async fn get_version(instance: &Instance) -> (String, i64) {
+    let res = instance
+        .debug_interface_client()
+        .get_node_metric("libra_state_sync_committed_version{}")
+        .await;
+    let content = match res {
+        Ok(res) => res.unwrap_or_default(),
+        _ => 0i64,
+    };
+    (instance.peer_name().clone(), content)
```

Comment:
I don't like that we had to change `Instance::validator_index` function to parse string returned by here.

I think, you can just return instance here and make return type `(&Instance, i64)`